src/SplitPackages/Program.cs

```diff
@@ -0,0 +1,288 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+
+namespace SplitPackages
+{
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            if (args.Contains("help", StringComparer.OrdinalIgnoreCase) ||
+                args.Length < 3 ||
+                args.Length > 4)
+            {
+                PrintUsage();
+                return;
+            }
+
+            bool whatif = args.Length == 4 && args[3].Equals("whatif", StringComparison.OrdinalIgnoreCase);
+
+            try
+            {
+                var arguments = NormalizeArguments(args);
+                var packagesFromSource = GetPackagesFromSourceFolder(arguments.SourceFolder);
+                var packagesFromCsv = GetPackagesFromCsvFile(arguments.CsvFile);
+                var expandedPackages = ExpandPackages(packagesFromCsv, arguments.SourceFolder);
+
+                EnsureNoExtraSourcePackagesFound(packagesFromSource, expandedPackages);
+
+                EmitWarningsForMissingPackages(packagesFromSource, packagesFromCsv);
+
+                EnsureDestinationDirectories(
+                    arguments.DestinationFolder,
+                    expandedPackages.Select(e => e.DestinationFolderName).Distinct());
+
+                SetDestinationPath(arguments, expandedPackages);
+
+                CopyPackages(expandedPackages, whatif);
+            }
+            catch (Exception e)
+            {
+                WriteError(e.Message);
+                return;
+            }
+        }
+
+        private static Arguments NormalizeArguments(string[] args)
+        {
+            var sourcePackages = Path.GetFullPath(args[0]);
+            var csvFile = Path.GetFullPath(args[1]);
+            var destinationFolder = Path.GetFullPath(args[2]);
+
+            if (!Directory.Exists(sourcePackages))
+            {
+                var msg = $@"Source packages folder does not exist or is not a folder
```

Comment:
is `@` required at all? Also why do you need `msg`?  Can't you just inline the message?